### DISCUSSION TITLE
Delete Images from Mobile Device and keep on Server

### DISCUSSION THREAD

**Discussion by Quadrubo** [immich-app/immich#2379] [Q&A]
Is there a way to delete the images form my mobile device and keep them stored on the server or is the mobile device the complete source of truth?

**Comment by SitramSoft** [comment]
I would also be interested in an answer to above question

**Comment by PeterBasista** [comment]
Just back it up (wait until it's in immich) and then delete it on your device (Via the device app and not via immich). Then it will only be an asset on the server.

**Reply by bcsteeve** [reply]
But that's not the behavior described in the app.  When deleting in the app, it warns you that it will delete from BOTH the device and the server.

**Reply by PeterBasista** [reply]
Sure. Sorry. I misunderestand the question.

**Reply by alextran1502** [reply]
This is the same behavior as GPhotos. There will be two steps after you confirm the dialog.

1. Delete from Immich server
2. The native dialog will pop up and ask if you want to delete that photo from the device as well - since this is the native behavior (removing file from file system) the OS make sure this is what you want to do and to prevent malicious app to remove content on your device

**Reply by jkumeboshi** [reply]
You understand that this workaround works, yes, but is completely useless since I've to find SAME exact photos on the device App, where are sorted in a completely different way. And that for EACH single photo I need to delete from device.

**Comment by Sherlock1979** [comment]
It would be awesome to have the ability to delete from device only (within the mobile app).

**Comment by bcsteeve** [comment]
Perhaps someone can help me understand this.  I've been setting up Immach (server) for a few weeks now, trying to get it to work with my 100k+ photo collection, and I think I'm close.  But just now I realized that when I delete an item on the server, it gets backed-up again from the device (which sort of makes sense as a backup-focused app) but ALSO that if I delete an item on the device, it removes it from the server (which does NOT adhere to basic backup principles).

I must be missing something.  How does one delete from the device without deleting from the server?

**Reply by PeterBasista** [reply]
Backup image back when you delete it from the server just make sense. But if you remove image from device native OS app then the server image will be still presented in immich (if was already processed by immich and backed up). If you delete image from immich app you are basically removing it from the server and then OS popup will show up asking you if you are sure to remove them from local device too.

And answer for your question. 
Use device native OS app to delete images just from device.

Immich is primary client for server images. So if you are handling images just from immich app then you are primary handling server stuff.

**Reply by alextran1502** [reply]
To delete from the device without deleting from the server, it has to happen from your native gallery app.

When the stable release roll out, we will add in a feature to "free up space" on your device. Since Immich is still in heavy development, we don't want to include this feature yet

**Reply by bcsteeve** [reply]
Excellent, that works, thank you.

**Reply by bcsteeve** [reply]
Hmm... OK, so I tried this.

I went into my phone's gallery app and deleted everything.  Then I went into the Immich app... and everything that was natively on my phone is now gone from immich too?    But it is still there on the server.  Here's a screenshot of what I see on my phone now:  https://photos.app.goo.gl/fTk6NN3J4nTTDcbc7

Note that the image from July 2022 was never on this phone.

What did I do wrong?  How do I get them showing up again in the app?

Edit:  Fixed it.  
 - Tried and failed:  force stop app and reload.
 - Tried and resolved:  logged out and logged back in on the app.

**Reply by bilogic** [reply]
Can I suggest that in the Settings, Backup, add 1 more checkbox, `Delete on device after backup`? This will address all our concerns.

**Reply by diegodcp** [reply]
Great @bcsteeve thanks!
I was searching for that, delete images from my phone's gallery and keep it on the server. But then I cannot see the photos in the immich app (as in you screenshot). 
Only with a simple logout and login, and images are shown correctly in the immich app, this is in 1.107.2 version.

Besides the checkbox (comment above), if we remove images from the phone gallery, it would be great if the update is automatically without a logout (with a swipe bottom or refresh button).

**Reply by bilogic** [reply]
frankly, after some further testing and research, it is much easier to use nextcloud. I'm not sure how `keeping multiple copies everywhere` and `read only external storage` is a thing, the whole idea of using an app is to manage data in a more efficient manner and immich does neither.

**Reply by alextran1502** [reply]
@bilogic hello, what is it that you are looking to do?

**Reply by bilogic** [reply]
To answer your question, I just want to manage my photos as a single user. Now that will sound generic and vague, so let me boil it down further

- As a default, when a file is changed (CRUD), I want it to also change everywhere that is connected to the app.. this is really to answer, why would most users need to host their own *photo* app?
- Keeping an offline copy and other use cases can be further added/configured as required
- We can further add on rules on how the app will work in multi user scenarios

The more immediate issues are

1. Single copy on cloud aka single source of truth (SSOT), I think this whole discussion and many others were asking for it with all sorts of vocab and descriptions over years. I mean, in my view, they all stem from this SSOT requirement
2. Read/write human manageable folder structure, nobody feels safe if their data cannot be easily recovered at a file system level and everybody has a different way to organize their files (and it is tremendously important to most, imagine I shift your files around to according to my structure)
3. immich's EXIF editing for captions/descriptions is fantastic, but it is greatly handicapped by a read-only external storage concept. I mean when it comes to photos, does it make sense to have such a concept? It's almost like plugging a USB drive into your machine only to realize you can read and not write, storage is storage, users don't care if it is internal or external

Personally, 
- if images can be managed like navidrome's music (a cloud music player), it would be much easier to conceptualize and understand
- 3rd party android clients can select files to be kept on device for offline access (while still respecting the cloud as the SSOT)

I have spent over an hour writing the above trying not to write off the cuff, I'm not sure if spending more time will influence the direction in a significant manner. If it gains any traction, maybe we can discuss this more systematically.

**Reply by winkmal** [reply]
Thanks for the tip, @bcsteeve 
> Tried and resolved: logged out and logged back in on the app.

Worked partially for me. The photos uploaded from the iPhone now show again, on the Immich iPhone app. :iphone: 
However, on the iPad and the browser view, I still get the blurred image with the error for the very same photos... :frowning_face: 
Any ideas on what else I could try? :thinking:

**Reply by alextran1502** [reply]
@winkmal this means your UPLOAD_LOCATION drive isn't mounted correctly. Can you double check?

**Reply by winkmal** [reply]
Dear Alex, thanks for the prompt reply. How could I check this? :thinking: 
I took the Docker compose file [from here](https://github.com/imagegenius/docker-immich?tab=readme-ov-file#docker-compose) and adjusted it to my needs. Til now, `UPLOAD_LOCATION` was not explicitly set. 
Now, I played around a bit and ended up with the following compose file (posting only the relevant part):
```yml
---
services:
  immich:
    image: ghcr.io/imagegenius/immich:latest
    container_name: immich
    environment:
      - PUID=1000
      - PGID=1000
      - TZ=Europe/Berlin
    env_file:
      - .env
    volumes:
      - ./config:/config
      - ${UPLOAD_LOCATION}:/photos # place where Immich App stores my uploaded photos from iPhone
      - /shared/backups/winkmal/pixelPhone:/import/winkmal/pixelPhone:ro
      - /shared/media/winkmal/Photos:/import/winkmal/Photos:ro # for desktop uploads
      - /etc/localtime:/etc/localtime:ro
    ports:
      - 2283:8080
    restart: always
    network_mode: bridge
```

And the corresponding `.env` file:
```yml
# You can find documentation for all the supported env variables at https://immich.app/docs/install/environment-variables

# The location where your uploaded files are stored
UPLOAD_LOCATION=/shared/backups/winkmal/iFotos
IMMICH_MEDIA_LOCATION=/photos
# The location where your database files are stored
DB_DATA_LOCATION=./postgres

# Connection secret for postgres. You should change it to a random password
DB_PASSWORD=postgres

# The values below this line do not need to be changed
###################################################################################
DB_USERNAME=postgres
DB_DATABASE_NAME=immich
```
Using `IMMICH_MEDIA_LOCATION=/photos` was [suggested here](https://github.com/imagegenius/docker-immich/issues/354#issuecomment-2102158091).
I also get lots of `offline paths` under `2283/admin/repair`, as reported by @wjfu99
Maybe I should mention that I use a different (non-admin) user on the iOS apps, who owns the libraries. I.e., my admin user does not have a single photo in his timeline.

Even after several container restarts, the problem still persists. Any more ideas?

**Comment by juxeii** [comment]
Any updates on this? Without this feature, Immich is not usable at all for me and familiy. My wife keeps on cleaning up her phone and she definetly wants to keep those on the server.

**Reply by alextran1502** [reply]
You can already remove your photos and library from your phone's native media viewer, and they won't be removed from the server.

**Reply by juxeii** [reply]
Sure, but Immich is supposed to be a one-stop-shop for photos and backups, isn't it?
Resorting to a another app for just local deleting stuff is not user friendly.
I can never tell my wife of doing such complicated things, more so when with Google Photos you can do it.

Do you have a feature id for "free up space" where we can follow progress?

**Reply by alextran1502** [reply]
Sure, there are still many missing features, so they will get implemented at some point. You can find the relevant PR here https://github.com/immich-app/immich/pull/4015

**Reply by Quadrubo** [reply]
#4015 is a different thing than what this discussion is about though right? Is there any free up space pr for example? where one get's asked if the image should stay on the server and is then deleted locally only?

**Reply by mix3d** [reply]
> Sure, there are still many missing features, so they will get implemented at some point. You can find the relevant PR here https://github.com/immich-app/immich/pull/4015

I feel like this PR is the OPPOSITE of what the thread is asking for. 

I'm, probably like many others, trying to use immich as a 100% Google photos replacement. Having to leave immich to achieve a goal seems counter intuitive.

What brought me here was searching for "how do remove the image from my device to free up device space but leave the images in my immich gallery and originals on the server", after struggling in the app to figure it out and accidentally hard deleting (from both device and server), so I'm curious where this sits. 

I'm ok with it being per-image instead of a generic (hopefully configurable) "free up space" feature, at least to start, ideally something that gives me per-album or source folder control.

**Reply by shenlong-tanwen** [reply]
> I'm ok with it being per-image instead of a generic (hopefully configurable) "free up space" feature, at least to start, ideally something that gives me per-album or source folder control.

You should already be able to do this to single or multiple assets from the main timeline. Select a group of asset and swipe the row of icons in the header of the bottom sheet and you should find an option to delete only the local part of the asset

**Comment by shenlong-tanwen** [comment]
PR for local only deletion in mobile app
- #4505

**Reply by bilogic** [reply]
![image](https://github.com/immich-app/immich/assets/946010/a044c0ed-3fda-4161-ae3b-f74c596c90b0)

How about adding `Move to server` or `Remove from device`?

**Comment by JoeeGrigg** [comment]
I'm having an issue here where managing the photos entirely in the Immich works perfectly. However, when I want to delete an image locally I will go and do it from the native iOS Photos app. That works and the image remains on the Immich server (visible on the web). The problem is, the Immich app then displays an error / photo missing icon for that image rather than just serving it from the Immich server. This only seems to be the case for photos marked as uploaded from this device and original file still on this device (a cloud with a tick) before I delete it.

Docs of cloud with a tick: https://immich.app/docs/FAQ/#what-is-the-difference-between-the-cloud-icons-on-the-mobile-app

![IMG_DF4E299D457E-1](https://github.com/immich-app/immich/assets/8267620/d377bde5-f0ab-4c15-9f6d-c22c77a1f185)

I would also be in huge favour of a "Free up space" feature which would delete the files locally from within the Immich app without having to use the native photos app.

**Reply by z1haze** [reply]
I've run into this same issue and it really kind of I'd a deal breaker.  The vast majority of us users use tools like this to keep our devices clean and our photos safe. If we cannot do simple things like backup and delete local, what's the point? 

That said, I have found that if you log out and log back in, sometimes it fixes the broken images in immich, but I think we all want to delete local images via the immich app without effecting the backup.

**Reply by JoeeGrigg** [reply]
Yes, I have found the same log out and back in again work around but it is a huge pain to have to deal with. I really have my fingers crossed that a "free up space" like feature comes soon...

**Reply by cfelicio** [reply]
I was having the same issue, there is a potential solution that does not require logging in and out. Go to settings -> advanced, and enable "prefer remote images". After doing this, I'm able to see everything, even after deleting from the phone app.

**Reply by richard015ar** [reply]
Thanks for this tip! ! It did the trick for me as well

**Reply by valerioformato** [reply]
+1 I'm having the same issue. I enabled "prefer remote images" and logged out. 

Will I have to do the same next time I delete content locally?

**Comment by mmomjian** [comment]
duplicate of #165, this will be added in/after stable release. thank you

**Reply by uncle-b** [reply]
Hi, Is there any sight on this functionality (deleting backed up images from device) coming to life soon? This is really holding me back from making the switch.

**Reply by Skyfay** [reply]
Yes, the same applies to me. I'm currently making the switch, but this feature is a must for me.

**Reply by bcsteeve** [reply]
Of course you can delete backed up images from the device while leaving it on the server.  That's what "backed up" means, and is the primary function of this software.  I personally find it SIMPLER to open a different gallery app and delete from there (which, perhaps, speaks to a UX concern), but you can just use the "delete from device" button.

**Reply by Skyfay** [reply]
> Of course you can delete backed up images from the device while leaving it on the server.  That's what "backed up" means, and is the primary function of this software.  I personally find it SIMPLER to open a different gallery app and delete from there (which, perhaps, speaks to a UX concern), but you can just use the "delete from device" button.

But we would like to delete it directly from the immich app. Synology did that really good:
![image](https://github.com/user-attachments/assets/37e8ab84-b725-42e0-bbc0-ff03bb745bb9)

**Reply by bcsteeve** [reply]
So do it right from the app then.  Am I missing something? I told you what button to press .  I'm on Android, so if this is an Apple problem, I don't know...

**Reply by mix3d** [reply]
The ask is to have immich _automatically_ remove backed up images from the device, so you don't have to manually clean up storage space. This isn't OS specific.

Ideally this is configurable by date, folder, type, etc.

Eg: remove backed up images older than 1 month from my /Photos folder

**Reply by Skyfay** [reply]
> So do it right from the app then.  Am I missing something? I told you what button to press .  I'm on Android, so if this is an Apple problem, I don't know...

Where? I have to select them all and then execute them. There is no menu where I can do that.

**Reply by mix3d** [reply]
The delete from device button is currently all the way at the end of the bottom menu, you have to scroll to the right.

**Reply by uncle-b** [reply]
I understand that you can manually delete images from the device one by one, but a typical user (such as myself) does not tend think about this until the device itself starts complaining about storage limitations. At that time, they (or I) may have 20000+ images on the device, which makes it quite a task to manually delete backed up images. That is what lies at the core of the question.

**Reply by Skyfay** [reply]
> The delete from device button is currently all the way at the end of the bottom menu, you have to scroll to the right.

But you have to do each one individually. I want to do them all with one click. My routine with Synology was to delete everything on my device once a month.

**Reply by bcsteeve** [reply]
> The ask is to have immich _automatically_ remove backed up images from the device, so you don't have to manually clean up storage space. This isn't OS specific.
> 
> Ideally this is configurable by date, folder, type, etc.
> 
> Eg: remove backed up images older than 1 month from my /Photos folder

That might be  YOUR ask, but that wasn't what was being asked.  Personally, I think that would be a terrible feature (here Dad, let me show you this picture of my kid when he was born... damn it, it was automatically pruned), but I do understand the use-case and maybe you should create a feature request for that.  THIS feature request, which is closed because it was a duplicate of another feature request WHICH WAS COMPLETED, is not the same thing.



> I understand that you can manually delete images from the device one by one, but a typical user (such as myself) does not tend think about this until the device itself starts complaining about storage limitations. At that time, they (or I) may have 20000+ images on the device, which makes it quite a task to manually delete backed up images. That is what lies at the core of the question.

I don't understand why people think that THEIR specific use-case represents "a typical user".  I'm not trying to pick on you... it is a SUPER common thread in these complains: "I want it exactly MY way and 99.9% of users want the same thing".  No, they don't.  The world is full of compromise and different setups and making something work for YOUR specific needs will break it for someone else (dare I say, the "typical" user? No, I dare not.).

But if that's your issue, then the obvious solution is to use a tool suited for the job.  Immich doesn't have to be everything for everyone.  Go into your favorite file manager and delete your files... happy in the knowledge that Immich *did its job* and backed them up.

> > The delete from device button is currently all the way at the end of the bottom menu, you have to scroll to the right.
> 
> But you have to do each one individually. I want to do them all with one click. My routine with Synology was to delete everything on my device once a month.

Ok, then a) keep using Synology if you liked it so much or b) do it from a different app.  Easy.





Folks:  The requested feature was asked for AND DELIVERED *a long time ago*.  If you have a SPECIFIC request to IMPROVE the EXISTING FEATURE, then think it through, word is appropriately (not "and I'm not going to switch until it is exactly like XYX"), and make a new feature request.

**Reply by Skyfay** [reply]
> I don't understand why people think that THEIR specific use-case represents "a typical user". I'm not trying to pick on you... it is a SUPER common thread in these complains: "I want it exactly MY way and 99.9% of users want the same thing". No, they don't. The world is full of compromise and different setups and making something work for YOUR specific needs will break it for someone else (dare I say, the "typical" user? No, I dare not.).

Bro, that's a basic function that almost every one of these programs offers. It has nothing to do with 99.9%. This whole answer is completely unnecessary. But whatever...

**Reply by mix3d** [reply]
@bcsteeve #165 at the top of this thread is literally what I said.

**Reply by uncle-b** [reply]
Nevermind, I'll stick to Google then. Like 99.9% of the world.

**Comment by JaydenWSiena** [comment]
Can you guys not argue with each other on a watched thread? You’re spamming
my inbox.
